### PR TITLE
Fix crash when ad notification displays on Android

### PR DESCRIPTION
**Pull request by emerick**
Fix https://github.com/brave/brave-browser/issues/8109

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_ads/browser/ad_notification.h [hunk 1]
```diff
@@ -23,7 +23,8 @@ struct NotificationInfo;
 
 namespace brave_ads {
 
-const char* const kBraveAdsUrlPrefix ="chrome://brave_ads/?";
+// On Android, this URL must represent an HTTP or HTTPS web origin.
+const char* const kBraveAdsUrlPrefix = "https://www.brave.com/ads/?";
```
Comment (emerick) [inline components/brave_ads/browser/ad_notification.h]: This was required in C80, I believe due to this changeset:

https://chromium-review.googlesource.com/c/chromium/src/+/1886890

**Comment by emerick** [comment]
CI passed, outside of unrelated problem with linter and network audit.

**Comment by gdregalo** [comment]
@emerick  Nevertheless it doesn't crash the app this fix is not correct. All the notifications are treated as  web notifications and sent through  Sites channel. Please let me know if you need help to look into this.

**Comment by emerick** [comment]
Spoke with @gdregalo and it sounds like this will require a slightly different fix for browser-android-tabs, but this fix does appear to work correctly on brave-core Android from what I can see (probably due to some of the notification redesign that occurred). If we arrive at a solution that works for both, we can revisit here too.

**Pull request merged**